app/src/main/java/com/example/currencyconverter/domain/repository/CurrencyRepository.kt

```diff
@@ -0,0 +1,77 @@
+package com.example.currencyconverter.domain.repository
+
+import com.example.currencyconverter.data.database.AppDatabaseDao
+import com.example.currencyconverter.data.network.CurrencyApi
+import com.example.currencyconverter.domain.Repository
+import com.example.currencyconverter.domain.model.Currency
+import com.example.currencyconverter.domain.model.CurrencyPair
+import com.example.currencyconverter.domain.model.CurrencyRates
+import com.example.currencyconverter.mappers.CurrencyDatabaseMapper
+import com.example.currencyconverter.mappers.CurrencyDomainMapper
+import com.example.currencyconverter.mappers.CurrencyNetworkMapper
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.withContext
+
+class CurrencyRepository(
+    private val database: AppDatabaseDao,
+    private val apiService: CurrencyApi,
+    private val databaseMapper: CurrencyDatabaseMapper,
+    private val networkMapper: CurrencyNetworkMapper,
+    private val domainMapper: CurrencyDomainMapper
+) : Repository {
+
+
+    override fun getRatesByBaseAcronym(acronym: String): Flow<List<CurrencyRates>> {
+        return database.getLatestRatesByBaseCurrencyAcronym(acronym).map {
```

Comment:
NIT: try using ctrl+alt+L to get convention-formatted code. You can use it for the entire file, or just select a code piece and then hit this hotkey